kani-compiler/src/kani_middle/transform/check_uninit/delayed_ub/instrumentation_visitor.rs

```diff
@@ -41,17 +38,14 @@ pub struct InstrumentationVisitor<'a, 'tcx> {
 }
 
 impl<'a, 'tcx> TargetFinder for InstrumentationVisitor<'a, 'tcx> {
-    fn find_next(
-        &mut self,
-        body: &MutableBody,
-        bb: BasicBlockIdx,
-        skip_first: bool,
-    ) -> Option<InitRelevantInstruction> {
-        self.skip_next = skip_first;
-        self.current = SourceInstruction::Statement { idx: 0, bb };
-        self.target = None;
-        self.visit_basic_block(&body.blocks()[bb]);
-        self.target.clone()
+    fn find_all(&mut self, body: &MutableBody) -> Vec<InitRelevantInstruction> {
+        for (bb_idx, bb) in body.blocks().iter().enumerate() {
+            self.current_instruction = SourceInstruction::Statement { idx: 0, bb: bb_idx };
+            self.visit_basic_block(bb);
+        }
+        // Push the last current target into the list.
+        self.targets.push(self.current_target.clone());
+        self.targets.clone()
```

Comment:
Do we need to clone this? You could make this function take `self` by value and return `self.targets`.

I'm assuming we don't want this function to be called multiple times, right?

Reply:
Correct, I'll change that. Thanks!